packages/secure_storage/amplify_secure_storage/android/src/main/kotlin/com/amazonaws/amplify/amplify_secure_storage/amplify_secure_storage/EncryptedKeyValueRepository.kt

```diff
@@ -5,39 +5,87 @@ package com.amazonaws.amplify.amplify_secure_storage.amplify_secure_storage
 
 import android.content.Context
 import android.content.SharedPreferences
+import androidx.annotation.VisibleForTesting
 import androidx.security.crypto.EncryptedSharedPreferences
 import androidx.security.crypto.EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV
 import androidx.security.crypto.EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
 import androidx.security.crypto.MasterKeys
+import java.io.File
+import java.util.UUID
 
 class EncryptedKeyValueRepository(
     private val context: Context,
     private val sharedPreferencesName: String,
 ) : KeyValueRepository {
 
+    private val sharedPreferences: SharedPreferences by lazy {
+        EncryptedSharedPreferences.create(
+            "$sharedPreferencesName.${getInstallationIdentifier()}",
+            MasterKeys.getOrCreate(MasterKeys.AES256_GCM_SPEC),
+            context,
+            AES256_SIV,
+            AES256_GCM
+        )
+    }
+
+    private val editor: SharedPreferences.Editor by lazy {
+        sharedPreferences.edit()
+    }
+
     override fun put(dataKey: String, value: String?) {
-        with(getSharedPreferences().edit()) {
+        with(editor) {
```

Comment:
This was part of the PR in Amplify-Android (https://github.com/aws-amplify/amplify-android/pull/2113). I attempted to keep the two implementations aligned as much as possible.

I think there may have been two reasons to do this:
1. Avoid reading from the file system each time. That also could have been achieved by caching the result from file system read rather than caching the editor though.
2. Small perf improvement as each call to request an editor does currently block. See: https://android.googlesource.com/platform/frameworks/base.git/+/master/core/java/android/app/SharedPreferencesImpl.java

I don't believe there would be any issue with reusing the editor. SharedPreferences caches the data in memory, but the Editor only caches the current edits until apply/commit is called. @tylerjroach - can you confirm reusing the editor is fine here?

Reply:
Makes sense - yeah all the editor methods are synchronized anyway - seems safe to use repeatedly.

Reply:
Yes, it is best to reuse the editor.